src/snowflake/snowpark/session.py

```diff
@@ -232,6 +233,12 @@
 _PYTHON_SNOWPARK_AUTO_CLEAN_UP_TEMP_TABLE_ENABLED = (
     "PYTHON_SNOWPARK_AUTO_CLEAN_UP_TEMP_TABLE_ENABLED"
 )
+
+# AST encoding.
+_PYTHON_SNOWPARK_USE_AST = "PYTHON_SNOWPARK_USE_AST"
```

Comment:
Should we consider naming this about the specific behavior rather than the generic implementation?  For example, may be we have 'PYTHON_SNOWPARK_SEND_AST_TO_SERVER' and 'PYTHON_SNOWPARK_EXECUTE_AST_ON_SERVER' (dependent on the former parameter) later on.

Reply:
I think there are two concerns here:

1.) encoding the AST as part of API invocation on the client (test with/without to not break existing Snowpark python client).
2.) sending the AST through the REST API to the server.

I kept the name here rather general (USE_AST) for this reason. Do you think we should have two flags here? I see the whole AST story potentially as one big, feature flag for which a more generic name might be better suited.

Reply:
Long term I would think the parameter goes away altogether (SNOWPARK_PYTHON_EXECTUTE_AST_ON_SERVER), when the client snowpark impl is removed then it doesn't make sense to have a way to say not execute on the server, that would just be the only thing it does.

However, during development and gradual roll out it will likely be useful to have a way to send AST to the server while separately controlling whether we decide to "execute" from the AST on the server side.  This could be a useful safety if we run into a situation where a customers' scenario is not working properly (for whatever reason, may require changes needed on the server side which are slower to roll out), so they have an option to not be blocked by continuing to run on the client.  At the same time, we can continue to send the AST to the server for validation testing and coverage reporting, etc. 

Regarding (1) one thing I notice is I we don't have try/catch around emit_ast code branches.  If we were to encounter an unhandled exception here we should still expect it to continue with the existing client side implementation.

Reply:
I think it's okay to keep the parameter named more generically, so we can reuse it for Phase 1 while we have a hybrid client. A simple alternative might be to turn the type into int, so we can express more with it – in case that becomes necessary in a future client.